src/Magnum/Shaders/Generic.h

```diff
@@ -483,7 +497,8 @@ template<> struct Generic<3>: BaseGeneric {
     typedef GL::Attribute<3, Vector4> Tangent4;
     typedef GL::Attribute<4, Vector3> Bitangent; /* also ObjectId */
     typedef GL::Attribute<5, Vector3> Normal;
-    /* 6, 7 reserved for vertex weights / bone IDs */
+    typedef GL::Attribute<6, Vector4> Weights;
+    typedef GL::Attribute<7, Vector4> JointIds;
```

Comment:
```suggestion
    typedef GL::Attribute<7, Vector4ui> JointIds;
```

Also, where's the 2D variant?

Reply:
Why wouldn't this work for 2D?

Reply:
Because it's defined only in the 3D specialization of the class ;)